scheduler/src/cook/mesos/constraints.clj

```diff
@@ -114,6 +114,18 @@
   (let [needs-gpus? (job-needs-gpus? job)]
     (->gpu-host-constraint job needs-gpus?)))
 
+(defrecord rebalancer-reservation-constraint [reserved-hosts]
+  JobConstraint
+  (job-constraint-name [this] (get-class-name this))
+  (job-constraint-evaluate
+    [this _ vm-attributes]
+    (let [target-hostname (get vm-attributes "HOSTNAME")]
+      [(not (contains? reserved-hosts target-hostname))
+       "Host was reserved for rebalancing"]))
```

Comment:
Nitpick: "Host is reserved for rebalancing`.

Reply:
Done